src/gui/spend.go

```diff
@@ -0,0 +1,256 @@
+					return
+				default:
+					shareFactor, err := strconv.ParseFloat(advancedSpend.HoursSelection.ShareFactor, 64)
+					if err != nil {
+						logger.Error(err)
+						wh.Error400(w, fmt.Sprintf("invalid share factor: %v", shareFactor))
+						return
+					}
+
+					if shareFactor < 0 {
+						logger.Warning("negative share factor")
+						wh.Error400(w, "share factor cannot be negative")
+						return
+					}
+				}
+
+			case "match_coins":
+			case "":
+				logger.Error("missing hours selection mode when type is auto")
+				wh.Error400(w, "missing hours selection mode when type is auto")
+				return
+			default:
+				logger.Errorf("invalid hours selection mode: %v", advancedSpend.HoursSelection.Mode)
+				wh.Error400(w, fmt.Sprintf("invalid hours selection mode: %v", advancedSpend.HoursSelection.Mode))
+				return
+			}
+		}
+
+		if advancedSpend.ChangeAddress == "" {
+			logger.Warning("missing change address")
+			wh.Error400(w, "missing change address")
+			return
+		}
+
+		changeAddr, err := cipher.DecodeBase58Address(advancedSpend.ChangeAddress)
+		if err != nil {
+			logger.Errorf("invalid change address: %v", err)
+			wh.Error400(w, fmt.Sprintf("invalid change address: %v", err))
+			return
+		}
+
+		// check whether destination addresses are correct or not
+		destList := make([]coin.TransactionOutput, len(advancedSpend.To))
+		for idx, to := range advancedSpend.To {
+			// check that the address is valid
+			toAddress, err := cipher.DecodeBase58Address(to.Address)
+			if err != nil {
+				logger.Errorf("invalid destination address %v", to.Address)
+				wh.Error400(w, fmt.Sprintf("invalid destination address %v", to.Address))
+				return
+			}
+
+			// convert coins to droplets
+			coins, err := droplet.FromString(to.Coins)
+			if err != nil {
+				logger.Errorf("unable to convert coins to droplet: %v", err)
+				wh.Error400(w, fmt.Sprintf("invalid coin amount %v", to.Coins))
+				return
+			}
+
+			destList[idx] = coin.TransactionOutput{
+				Address: toAddress,
+				Coins:   coins,
+			}
+
+			// parse coinhours
+			// when mode is auto the Hours field can be empty
+			// when mode is manual Hours field cannot be empty
+			if to.Hours != "" {
+				hours, err := strconv.ParseUint(to.Hours, 10, 64)
+				if err != nil {
+					logger.Errorf("unable to parse coinhours: %v", err)
+					wh.Error400(w, fmt.Sprintf("invalid coinhours %v", to.Hours))
+					return
+				}
+
+				destList[idx].Hours = hours
+			} else if advancedSpend.HoursSelection.Type == "manual" {
+				logger.Errorf("coinhours value missing for %v when mode is manual", to.Address)
+				wh.Error400(w, fmt.Sprintf("coinhours value missing for %v when mode is manual", to.Address))
+				return
+			}
+
+		}
+
+		// create a wltmap
+		wltMap := make(map[string]struct{})
+		for _, wlt := range advancedSpend.Wallets {
+			wltMap[wlt] = struct{}{}
+		}
+
+		// fetch all wallets on the system
+		wallets, err := gateway.GetWallets()
+		if err != nil {
```

Comment:
there are non-500 errors like ErrWalletAPIDisabled; see the `switch err` used in other wallet handlers

Reply:
done